pynet/datasets/lpsnc.py

```diff
@@ -0,0 +1,206 @@
+
+    def _crop(arr):
+        # return arr[45: 195, 30: 220, 10: 145]
+        return arr
+
+    def _norm(arr):
+        arr = np.nan_to_num(arr)
+        logical_mask = (arr != 0)
+        mean = arr[logical_mask].mean()
+        std = arr[logical_mask].std()
+        return ((arr - mean) / std).astype(np.single)
+
+    def merge_rois(roifiles, v=1):
+        """ Merge the input ROIs.
+
+        Parameters
+        ----------
+        roifiles: list of str
+            the ROI images to be merged.
+        outdir: str
+            the destination folder.
+        fmaskname: str
+            the filename of the mask to be saved.
+
+        Returns
+        -------
+        merged_file: str
+            the merged ROI.
+        """
+        # Check inputs
+        if len(roifiles) == 0:
+            raise ValueError("Expect at least one ROI.")
+
+        # Merge ROIS
+        im = nib.load(roifiles[0])
+        ref_affine = im.affine
+        merged_data = np.nan_to_num(im.get_data())
+        if merged_data.ndim != 3:
+            raise ValueError("Expect 3d-ROIS.")
+        if len(roifiles) > 1:
+            for path in roifiles[1:]:
+                im = nib.load(path)
+                affine = im.affine
+                if not np.allclose(ref_affine, affine):
+                    raise ValueError(
+                            "ROI files have different affine matrices.")
+                data = np.nan_to_num(im.get_data())
+                if data.ndim != 3:
+                    raise ValueError("Expect 3d-ROIS.")
+                merged_data += data
+        merged_data[merged_data > 0] = v
+        return merged_data
+
+    mapping_path = os.path.join(datasetdir, "data.json")
+    zero_path = os.path.join(datasetdir, "zero.nii.gz")
+    # path for empty image of correct dimensions
+    zero = nib.load(zero_path).get_data()
+    if not os.path.isfile(mapping_path):
+        raise ValueError(
+            "Are you in the right folder? "
+            + "Your folder= '{0}' ".format(datasetdir)
+            + "You may need special access for LPSNC dataset")
+    desc_path = os.path.join(datasetdir, "pynet_lpsnc_"+str(modality)+".tsv")
+    input_path = os.path.join(datasetdir,
+                              "pynet_lpsnc_inputs_"+str(modality)+".npy")
+    output_path = os.path.join(datasetdir,
+                               "pynet_lpsnc_outputs_"+str(modality)+".npy")
+    if not os.path.isfile(desc_path):
+        df = pd.read_json(mapping_path)
+
+        arr = df[["sub", "ses"]].values
+        input_dataset = []
+        output_dataset = []
+        nb_subjects = len(arr)
+        arr1 = []
+        with progressbar.ProgressBar(max_value=nb_subjects,
+                                     redirect_stdout=True) as bar:
+            for cnt, (sub, ses) in enumerate(arr):
+                logger.debug("Processing {0}...".format(sub))
+
+                subdata = df.loc[df["sub"] == sub]
+                if (modality == 0):
```

Comment:
Group the if else code by defining the modalities parameter that is set to MODALITIES in case no modality selection is specified, [modality] otherwise.